### ISSUE TITLE
Unable to update a failed application upgrade "can not upgrade application now"

### ISSUE THREAD

**Issue by Menkveld-24** [kubesphere/kubesphere#5525]
**Describe the Bug**
Tldr; I wanted to upgrade an application from the openpitrix store but made a mistake in the new Target App Settings and now I am unable to upload new settings.

I wanted to update Kasten (from openpitrix) from 5.0 to 5.5.5. I did this by selecting the app -> more -> edit settings.
<img width="382" alt="Schermafbeelding 2023-02-14 om 17 49 11" src="https://user-images.githubusercontent.com/43108191/218803036-f1020e13-8e96-4df7-a6ee-9faa9ac0c407.png">

I selected the proper target version and applied modifications to the config. Unfortunately I made an error in the config and this set the application status to Failed.
<img width="355" alt="Schermafbeelding 2023-02-14 om 17 46 35" src="https://user-images.githubusercontent.com/43108191/218802341-f00e104d-ea06-43bc-8042-0668eefae145.png">

After the failed update I tried the steps again but this gives me the error: 
<img width="438" alt="Schermafbeelding 2023-02-14 om 17 49 59" src="https://user-images.githubusercontent.com/43108191/218803261-9cf1e88b-ce80-4114-a833-1f054ef9c328.png">

However the current app version is set to 5.5.5 (targeted version) but the application that is still running is at 5.0 (original version)

**Versions Used**

KubeSphere: 3.3.0
Kubernetes: (If KubeSphere installer used, you can skip this)

**Environment**

Ubuntu Server 22.04 / 1 master/node: 12cpu/32g

**How To Reproduce**
1. Go to into a project that was deployed from the app store
2. application workloads -> apps -> click app
3. fold open "More" -> edit settings
4. Apply a new version with a configuration error and press ok
5. Wait for the upgrade to fail
6. Try steps 3-4 again and you are unable to update

**Expected behavior**
To be able to update the config when a mistake was made.
